plugins/catalog-backend-module-incremental-ingestion/api-report.md

```diff
@@ -33,8 +33,8 @@ export type EntityIteratorResult<T> =
 // @public (undocumented)
 export class IncrementalCatalogBuilder {
   // (undocumented)
-  addIncrementalEntityProvider<TCursor, TContext>(
-    provider: IncrementalEntityProvider<TCursor, TContext>,
+  addIncrementalEntityProvider<TCursor, TContext, TInput>(
+    provider: IncrementalEntityProvider<TCursor, TContext, TInput>,
```

Comment:
I'll see how complicated it will be to switch to `unknown` for all of them. That would certainly make some things simpler.

I do really like the idea of making the entity provider able to handle creation/deletion/renaming. Is it appropriate for the entity _provider_ to do that, though? It seems like it might be a little out of scope.

Reply:
I'm thinking a github discovery provider that you just point at an entire org. Whenever people make new repos in there, those should also be part of the discovery. And when repos are deleted, the corresponding entities should vanish.

Reply:
That would be handled by the delta mechanism. Deltas can add, remove, and change entities. I thought you were suggesting that the provider actually do the work of add/remove/change, instead of processing deltas from the data source.

Reply:
I think this is fully addressed by the engine, actually. It accepts a payload of both emitted entities and of deleted entities.